added docs for BukkitSchedulerMock (#1782)
BetonQuest/BetonQuest: docs/Participate/Process/Code/Writing-JUnit-Tests.md

    The `log` is a new instance of the `BetonQuestLogger` that you can use to log things during the test.
    This logger has a topic that can be accessed via `BetonQuestLoggerService.LOGGER_TOPIC`.

## Handling BukkitScheduler

If you want to test code that only works with the `BukkitScheduler`, we even have a ready to use solution for this.
To use the `BukkitSchedulerMock` you need to create the following setup:

````java linenums="1" hl_lines="3-5"
@Test
void testMethod {
    final BukkitSchedulerMock scheduler = new BukkitSchedulerMock();
    try (MockedStatic<Bukkit> bukkit = mockStatic(Bukkit.class)) {
        bukkit.when(Bukkit::getScheduler).thenReturn(scheduler);
        ...
    }
}
````

> Now you can use the scheduler object to several things. First if you want to perform a single tick or multiple ticks,

Reviewer: ```suggestion
Now you can use the scheduler object for several things. First if you want to perform a single or multiple ticks,
```

you can call the methods `performTick()` or `performTicks(long)`:

````java linenums="1"
    scheduler.performTick();
    scheduler.performTicks(20);
````

We also have a method that allows to get the number of ticks since the `BukkitSchedulerMock` was created.

````java linenums="1"
    scheduler.getCurrentTick();
````

> There are some more methods that allows to do some more things with this scheduler:

Reviewer: ```suggestion
There are some additional features of this scheduler:
```
